### PR TITLE
Added demo module to show explicit imports of macros

### PR DESCRIPTION
**Pull request by gautamdhameja**
Now that we have to explicitly import `decl_module`, `decl_storage` and `decl_event` macros in the runtime modules, maybe adding a demo module to the node template can make it more ready for hacking.

### CODE REVIEW

**Comment by xlc** [comment]
There is already example module. Maybe just move it to node-template?
https://github.com/paritytech/substrate/blob/master/srml/example/src/lib.rs

**Comment by gautamdhameja** [comment]
Yes, the plan is to move the template module instead.
https://github.com/paritytech/substrate/blob/gav-template/srml/template-module/src/lib.rs
Will update the PR with these changes.

**Comment by gautamdhameja** [comment]
[clabot:check]

File: node-template/runtime/src/demo.rs [hunk 1]
```diff
@@ -0,0 +1,31 @@
+/// module for demonstration purposes and with necessary imports
+
+/// in order to create your runtime module, you need `decl_module` macro imported
+/// import `decl_storage` to declare and initialize storage for your module
+/// both `decl_module` and `decl_storage` are imported in line 7 below
+
+/// additionally import `decl_event` to declare events for your module
+
+/// feel free to remove or edit this file, as needed
+/// in case you remove this file, also make sure to remove line numbers 42, 178, 194 from `../runtime/src/lib.rs`
```
Comment (bkchr) [inline node-template/runtime/src/demo.rs]: This comment will break, just remove it.

**Pull request merged**